src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/deconvolution/deconvolution_kernel_b_fs_zyx_fsv16.cpp

```diff
@@ -283,4 +275,14 @@ JitConstants DeconvolutionKernel_b_fs_zyx_fsv16::GetJitConstants(const deconvolu
     return jit;
 }
 
+KernelsData DeconvolutionKernel_b_fs_zyx_fsv16::GetKernelsData(const Params& params, const optional_params& options) const {
+    KernelsData kds = Parent::GetKernelsData(params, options);
+
+    const deconvolution_params& orgParams = static_cast<const deconvolution_params&>(params);
+    if (orgParams.inputs[0].Feature().v % 16 != 0) {
+        kds[0].can_reuse_memory = false; // Set memory_reuse = false when input feature size is not 16 aligned.
```

Comment:
Why? Even if pad area contains non-zero values, deconv is supposed to multiply it by zeros from weights buffer

Reply:
Ticket 86502 issue happens in only Windows environment. The pad area contains NaN (0xffff is decoded to NaN in fp16) and it couldn't be ignored by zero multiplication. I also tried if() to check NaN but it affected to performance a lot.

Reply:
In that case, many other kernels may suffer from similar effects as well, so disabling of reuse for 2 kernels only is not an ultimate solution.
Have you checked why pad area contains NANs on windows? Is it written by some kernel or these buffers are not filled with zeroes, and NANs are located there right after buffer alloc?

Reply:
The NaN issue in Windows happens from 2nd iteration. In 1st iteration, it doesn't happen. I have a capture about comparison of input memory dump between 1st and 2nd. You can see pattern of NaN values in memory. The input data has feature size = 8.

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr'>



<PS terminal on   VSCode - i7-8700k>          <VS debugging   console - windows terminal - i7-8700k>          <Comparison of   memory dump 0xFFFFB80201960000 between 1st and 2nd iteration>          <<mem_1st.txt>>       <<mem_2nd.txt>>
--




</div>

<!--EndFragment-->
</body>

</html>

Reply:
Still not clear which code modifies the buffer

Reply:
Actually hard to find it. In Ubuntu, the NaN issue doesn't happen.

Reply:
>> In that case, many other kernels may suffer from similar effects 
I agree. However hard to find. :( and we needed a solution to avoid these observed issues without any perf loss.

Reply:
And but, even though the values are not Nan, there are kernels which is not using weights such as reduce. In that case, we'll get the perf degradation if we change the bahvior to deal with trash value.

Reply:
> And but, even though the values are not Nan, there are kernels which is not using weights such as reduce. In that case, we'll get the perf degradation if we change the bahvior to deal with trash value.

It'd be great to do an experiment with reduce op and check perf impact from partial reads and memory usage impact from disabling buffer reuse.

Reply:
One model related with reduce op is corel_nnart_01. Performance impact is 9% between partial read and whole.
![image](https://user-images.githubusercontent.com/1720339/180214260-d62eef6e-51bd-4225-b2af-12ff5b749ddc.png)

Reply:
@vladimir-paramuzov Done the experiments. Max mem increasing is 20MB in person vehicle bike detection. Performance geomean from partial reads is 1.000556187. If you need detail information, please let me know.

Reply:
@vladimir-paramuzov Could you please confirm the experiments result? Thanks!

Reply:
Seems to be acceptable.